pkg/plugin/keyvault.go

```diff
@@ -101,29 +112,44 @@ func newKeyVaultClient(
 		keyVersion:       keyVersion,
 		vaultURL:         *vaultURL,
 		azureEnvironment: env,
+		keyID:            fmt.Sprintf("%x", sha256.Sum256([]byte(fmt.Sprintf("%s/keys/%s/%s", *vaultURL, keyName, keyVersion)))),
+		algorithm:        kv.RSA15,
 	}
 	return client, nil
 }
 
-func (kvc *keyVaultClient) Encrypt(ctx context.Context, cipher []byte) ([]byte, error) {
+func (kvc *keyVaultClient) Encrypt(ctx context.Context, cipher []byte) (*service.EncryptResponse, error) {
 	value := base64.RawURLEncoding.EncodeToString(cipher)
 
 	params := kv.KeyOperationsParameters{
-		Algorithm: kv.RSA15,
+		Algorithm: kvc.algorithm,
 		Value:     &value,
 	}
 	result, err := kvc.baseClient.Encrypt(ctx, kvc.vaultURL, kvc.keyName, kvc.keyVersion, params)
 	if err != nil {
 		return nil, fmt.Errorf("failed to encrypt, error: %+v", err)
 	}
-	return []byte(*result.Result), nil
+
+	annotations := map[string][]byte{
+		"kms.plugin.io.date":                 []byte(result.Header.Get("Date")),
```

Comment:
Maybe use `azure.akv.io` as prefix for the annotation? That way if there is every a different plugin, they could just use `azure.<external kms>.io`. I'm open to other name suggestions for this!